### ISSUE TITLE
[Godot Physics] Simulating rigidbody composed of multiple collision shapes gets all its coordinates set to NaN

### ISSUE THREAD

**Issue by e344fde6bf** [godotengine/godot#47476]
**Godot version:**
3.3rc6,  3.3.rc.custom_build.cca2637b9

Couldn't reproduce this on 3.2.3

**OS/device including version:**
N/A

**Issue description:**

If you simulate a rigibody that uses multiple collision shapes, after some time all it's position and velocity values will be set to NaN, and the console will spam error messages:

```
ERROR: set_axis_angle: The axis Vector3 must be normalized.
   At: core/math/basis.cpp:980.
ERROR: instance_set_transform: Condition "Math::is_nan(v.x)" is true.
   At: servers/visual/visual_server_scene.cpp:723.
```

This only happens using the Godot physics backend. 

If you turn on drawing of collision shapes, you can see that it's drawing contact points between the rigid bodies own collision shapes:

![contact-points-own-collision-shapes](https://user-images.githubusercontent.com/79001820/112913392-dacf0e00-913c-11eb-9587-522defc67577.png)

**Steps to reproduce:**

* Create a rigidbody that is composed of multiple collision shapes
* Simulate it for some time and wait for the error messages.  The simulation is not deterministic, so it doesn't always happen.

**Minimal reproduction project:**
[rigid-compound-shape-bug.zip](https://github.com/godotengine/godot/files/6225215/rigid-compound-shape-bug.zip)

**Comment by akien-mga** [comment]
I can reproduce the issue, it seems to be have introduced between 3.2.4 beta 5 and 3.2.4 beta 6.

Changelog between the two: https://github.com/godotengine/godot/compare/a18df71789a36b318c40d691efdb4da1e574bbfd...7e207cfd48d6077ac6aaa3c45423d3fcf2f90bd7

From that list #44901 seemed like a potential candidate, and indeed disabling `3d/godot_physics/use_bvh` in the project settings fixes the regression for this project. CC @lawnjelly @pouleyKetchoupp

**Comment by qarmin** [comment]
Finally I found that `nan` and `inf` values are generated because in this line
https://github.com/godotengine/godot/blob/5fdb8b02e33fbf25f348583c1312aba648c3a7e1/servers/physics/collision_solver_sat.cpp#L339
`den` is equal to 0

**Comment by lawnjelly** [comment]
Sorry it's taken a while to get to this.

It looks like there's a check I've missed in the BVH .. if the calling userdata is the same for 2 colliding objects, the callbacks should not be sent (the userdata being the physics object, the table I guess?), which must be the mechanism used to prevent self collision. This should be easy enough to fix.

The spam from the physics may be this self intersection situation it cannot deal with for some reason.

**Comment by akien-mga** [comment]
Fixed by #47614.
